Create sample sites reference page

eiman-eltigani-ttd opened the pull request.

eiman-eltigani-ttd commented on sidebars.js:
```diff
@@ -1,528 +1,527 @@
+        items: [], 
+      },
+    ],
+  },
```
Here is the code I directly added (the rest is just formatting changes I made to make it easier to read/edit)

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+# UID2 Sample Sites
+
+This page lists all official UID2 sample implementations, with links to code, live demo sites, related documentation, and common use cases. Use this page to quickly find, compare, and switch between sample integrations that match your setup.
+
```
would add "may" in front of "match your setup."

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+Sample pages for publishers integrating UID2 directly into their websites using JavaScript or server-side logic. For more details, see [Publisher Web Integration Overview](../guides/integration-options-publisher-web.md).
+
+#### UID2 CSTG with JavaScript SDK
```
Actually I think this is supposed to be called "UID2 Client-Side Integration with JavaScript SDK" or "UID2 Client-Side with JavaScript SDK".  it is using CSTG but the type name of the site should be client-side

eiman-eltigani-ttd replied:
Agreed, I didn't include CSTG in the header or site name but updated description to mention it

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+#### UID2 Client-Server JS SDK (no CSTG)
+
+- Site: [UID2 Publisher Standard Integration Example](https://example-jssdk-integ.uidapi.com/)
```
this site needs to be renamed eventually on the actual site, but the name here would be "Client-Server" instead of "publisher standard".  I think that is an outdated name

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+#### UID2 CSTG with JavaScript SDK
+
+- Site: [UID2 Publisher Client-Side Integration Example using UID2 JavaScript SDK](https://cstg-integ.uidapi.com/)
```
we should take out "Publisher" from places here (at some point actually removing it from the site itself).  that's a bit outdated because now, there are other types of participants who also use it - advertisers and data providers also

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Site: [UID2 Publisher Client-Side Integration Example (JS SDK + Secure Signals)](https://secure-signals-client-integ.uidapi.com)
+- Code: [`uid2-web-integrations/examples/google-secure-signals-integration/client_side`](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/google-secure-signals-integration/client_side)
+- Docs: [Secure Signals Integration Guide](../guides/integration-google-ss.md)
+- Use Case: For publishers who want front-end only integration with GAM Secure Signals.
```
should say "client-side" instead of "front-end" to keep consistent

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+#### React Client-Side Secure Signals
+
+- Site: [UID2 Publisher Client-Side React Integration Example](https://secure-signals-react-integ.uidapi.com/)
```
nit: write "React Client-Side" instead of "Client-Side React" for consistency

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+#### Hybrid Client-Server Secure Signals (JS SDK v3)
+
+- Site: [UID2 Publisher Hybrid Integration Example (Server + JS SDK + Secure Signals)](https://secure-signals-client-server-integ.uidapi.com/)
+- Code: [`uid2-web-integrations/examples/google-secure-signals-integration/with_sdk_v3`](https://secure-signals-client-server-integ.uidapi.com/)
```
the code link here isn't right

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+  - `secure-signals-jssdk-integ.uidapi.com`
+  - `esp-jssdk-integ.uidapi.com`
+- Use Case: For publishers who want a hybrid approach (token generation on the server, client SDK logic for secure signals).
+
```
client-side logic instead of client SDK logic

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Additional Site Domains:
+  - [https://secure-signals-srvonly-integ.uidapi.com](https://secure-signals-srvonly-integ.uidapi.com)
+  - [https://esp-srvonly-integ.uidapi.com](https://esp-srvonly-integ.uidapi.com)
+- Use Case: For publishers who want all Secure Signals logic server-side for better control and security.
```
add "and UID2" after "all Secure Signals"

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+  - [https://esp-srvonly-integ.uidapi.com](https://esp-srvonly-integ.uidapi.com)
+- Use Case: For publishers who want all Secure Signals logic server-side for better control and security.
+
+#### Client-Server Secure Signals (JS SDK v3)
```
don't need to say "JS SDK v3" in the title here. it may not always be v3

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+Sample pages for adding UID2 support to Prebid.js header bidding setups. For more details, see [Prebid Integration Overview](https://unifiedid.com/docs/guides/integration-prebid).
+
+#### UID2 CSTG with Prebid.js
```
this should probably be "UID2 Client-Side Integration with Prebid.js" because the CSTG acronym hasn't been defined on this page

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+- Site: [UID2 React Client-Side Integration Example with Google Secure Signals](https://secure-signals-react-integ.uidapi.com/)
+- Code: [`uid2-web-integrations/examples/google-secure-signals-integration/react_client_side`](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/google-secure-signals-integration/react_client_side)
+- Docs: [Secure Signals Sample Implementation](https://unifiedid.com/docs/guides/integration-google-ss#sample-implementations)
```
was it intentional for the docs link here to be different than the docs links for the other secure signals implementations?

ashleysmithTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+### Web Integrations
+
+Sample pages for publishers integrating UID2 directly into their websites using JavaScript or server-side logic. For more details, see [Publisher Web Integration Overview](https://unifiedid.com/docs/guides/integration-options-publisher-web).
```
even though the docs link does go to something for publishers, make the title on the link just "Web Integration Overview"

genwhittTTD commented:
@ashleysmithTTD I definitely don't think it belongs at the top level, but I totally agree that in the reference section it'd have low visibility. In the Guides section also actually.
One related question is who's the audience... publishers, or other audiences as well?
I think it should go in reference but we should link to it from some of the high-level docs, a short section stating that there are several sample implementations available for testing, with a link to the doc... something like that. For example we could add it in the publisher overview and any other overviews.
And, even, in each section that references one of the sample implementations, we could add a line at the bottom of that section: For a full list of the sample implementations that are available, see [link]. Something like that.
So it would have good visibility that way, at the point in the docs where the implementation decision is being considered.
We could also add it in Publisher Integration Resources at the bottom. Anywhere it seems appropriate, just a one-liner and a link (with a heading in some cases, such as that one).

eiman-eltigani-ttd commented:
> @ashleysmithTTD I definitely don't think it belongs at the top level, but I totally agree that in the reference section it'd have low visibility. In the Guides section also actually. One related question is who's the audience... publishers, or other audiences as well? I think it should go in reference but we should link to it from some of the high-level docs, a short section stating that there are several sample implementations available for testing, with a link to the doc... something like that. For example we could add it in the publisher overview and any other applicable overviews, depending on the answer to the audience question. And, even, in each section that references one of the sample implementations, we could add a line at the bottom of that section: For a full list of the sample implementations that are available, see [link]. Something like that. So it would have good visibility that way, at the point in the docs where the implementation decision is being considered. We could also add it in Publisher Integration Resources at the bottom. Anywhere it seems appropriate, just a one-liner and a link (with a heading in some cases, such as that one). @eiman-eltigani-ttd FYI.

I like the idea of having it in the reference section and linking it at a higher level doc, and was also thinking the UID2 Integration Guides Summary page instead of the Publisher Overview since advertisers and data providers can also do cleint-side integration. Also want to update the title to something more straight forward since we're taking it out of context like UID2 Demo Sites or UID2 Live Examples.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+#### Client-Server Integration using UID2 JavaScript SDK
+
+- Site: [UID2 Client-Server Integration Example](https://example-jssdk-integ.uidapi.com/)
+- Code: [`uid2-web-integrations/examples/js-sdk`](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/js-sdk)
```
Does it have a title? If not, better to give the full URL in the copy, not a partial URL.

eiman-eltigani-ttd replied:
My thought was the shorter repo path makes it clear it's a code repository link compared to the URLs used for the live demos, but lmk what you think!

genwhittTTD replied:
I think that since it says Code, it's going to be clear that it's a code repo path. I'm inclined to have the full path but no worries if you want it like this. But we need to not have the code tag in the link.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Docs: [Publisher Integration Guide, Server-Side](https://unifiedid.com/docs/guides/integration-publisher-server-side)
+- Use Case: For publishers who want all UID2 logic handled server-side (no client SDK). Good for environments with strict client limitations or security requirements.
+
+---
```
What is this for?

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+### Prebid.js Integrations
+
+Sample pages for adding UID2 support to Prebid.js header bidding setups. For more details, see [Prebid Integration Overview](https://unifiedid.com/docs/guides/integration-prebid).
```
Make into a complete sentence.
For details (not more details)

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+### Google Secure Signals Integrations
+
+Sample pages for passing UID2 tokens to Google Ad Manager (GAM) via Secure Signals. For more details, see [Google Ad Manager Secure Signals Guide](https://unifiedid.com/docs/guides/integration-google-ss).
```
Same comments as previous intros.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+Sample pages for passing UID2 tokens to Google Ad Manager (GAM) via Secure Signals. For more details, see [Google Ad Manager Secure Signals Guide](https://unifiedid.com/docs/guides/integration-google-ss).
+
+#### Client-Side Secure Signals
+
```
We need an intro here before the list.

ashleysmithTTD replied:
Why do we need intros here for each of the secure signals sites? These are all just the headings for the site right under each one

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Use Case: For publishers who want a purely client-side integration with GAM Secure Signals.
+
+#### React Client-Side Secure Signals
+
```
We need an intro here before the list.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Use Case: For publishers with React apps who want to integrate UID2 + Secure Signals directly into their component-based architecture.
+
+#### Server-Side Secure Signals
+
```
We need an intro here before the list.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Use Case: For publishers who want all Secure Signals and UID2 logic server-side for better control and security.
+
+#### Client-Server Secure Signals 
+
```
We need an intro here before the list.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+---
+
+### Utilities
+
```
We need an intro here.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+### Utilities
+
+#### UID2 Hashing Tool
+
```
We need an intro here before the list.

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+- Docs: [Normalization and Encoding](https://unifiedid.com/docs/getting-started/gs-normalization-encoding)
+- Use Case: For developers or clients validating data preparation, such as hashing and normalization of emails before token requests.
+
+---
```
what is this?

genwhittTTD commented on docs/guides/integration-sample-sites.md:
```diff
@@ -0,0 +1,108 @@
+
+This page lists all official UID2 sample implementations, with links to code, live demo sites, related documentation, and common use cases. Use this page to quickly find, compare, and switch between sample integrations that may match your setup.
+
+---
```
This line is extra also. Remove any extra lines in the doc (in case I missed any others).

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+
+This sample is for publishers who prefer a fully client-side integration using the UID2 JavaScript SDK to generate and manage tokens directly in the browser. This approach is ideal for fast prototyping with minimal back-end requirements. The sample integration is available at the following links:
+
+- Site: [UID2 Client-Side with JS SDK Example](https://cstg-integ.uidapi.com/)
```
Copy should match the title of the link destination. In this case:

UID2 Publisher Client-Side Integration Example using UID2 JavaScript SDK

(aargh we need to fix that title re the SDK name! Sorry Eiman that this isn't cleaner for you.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+This sample is for publishers who prefer a fully client-side integration using the UID2 JavaScript SDK to generate and manage tokens directly in the browser. This approach is ideal for fast prototyping with minimal back-end requirements. The sample integration is available at the following links:
+
+- Site: [UID2 Client-Side with JS SDK Example](https://cstg-integ.uidapi.com/)
+- Code: [`uid2-web-integrations/examples/cstg`](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/cstg)
```
Don't use the code tag in a link (company internal style guidelines and it makes it not look like a link).

Same for all code links.

genwhittTTD commented on i18n/ja/docusaurus-plugin-content-docs/current/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+
+#### UID2 Server-Only (Server-Side)
+
+This sample is for publishers who want all UID2 operations to occur on the server-side, offering maximum control, security, and flexibility without relying on a client-side SDK. The sample integration is available at the following links:
```
on the server-side > on the server side

server-side is hyphenated as a modifier (server-side implementation) but not otherwise (on the server side).

(Same with client-side / on the client side.)

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+
+### Prebid.js Integrations
+
+This section includes a sample implementation demonstrating how organizations can integrate with UID2 to generate tokens and pass UID2 identity data into header bidding auctions. For details, see [Prebid Integration Overview](https://unifiedid.com/docs/guides/integration-prebid).
```
Link copy should match link destination. In this case:

UID2 Integration Overview for Prebid

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+- Code: [`uid2-examples/publisher/server_only`](https://github.com/IABTechLab/uid2-examples/tree/main/publisher/server_only)
+- Docs: [Publisher Integration Guide, Server-Side](https://unifiedid.com/docs/guides/integration-publisher-server-side)
+
+### Prebid.js Integrations
```
It's unnecessarily repetitive to have a major heading saying "this section includes..." and then only one smaller heading saying basically the same.

There might be more Prebid.js samples in the future no doubt, but for the moment I recommend combining this into one. Something like this:

### Prebid.js Client-Side Integration

This sample is for publishers who want to generate UID2 tokens on the client side and pass them into header bidding auctions using Prebid.js. The sample integration is available at the following links:

- Site: [UID2 Prebid.js Client-Side Integration Example](https://unifiedid.com/examples/cstg-prebid-example/)
- Code: [`uid2docs/static/examples/cstg-prebid-example`](https://github.com/IABTechLab/uid2docs/tree/main/static/examples/cstg-prebid-example)
- Docs: [Client-Side Integration Guide for Prebid.js](https://unifiedid.com/docs/guides/integration-prebid-client-side)

It would have to be changed back at some point perhaps -- you have the right structure -- but you're correct that it's unnecessarily repetitive when there is only one.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+- Site: [UID2 Server-Side Integration Example with Google Secure Signals](https://secure-signals-server-side-integ.uidapi.com/login)
+- Code: [`uid2-web-integrations/examples/google-secure-signals-integration/server_side`](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/google-secure-signals-integration/server_side)
+- Docs: [Secure Signals Integration Guide](https://unifiedid.com/docs/guides/integration-google-ss)
+- Additional Site Domains:
```
Additional Site Domains > Additional site domains

(all instances)

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,109 @@
+- Docs: [Normalization and Encoding](https://unifiedid.com/docs/getting-started/gs-normalization-encoding)
+
+:::note
+The Sample Sites above highlight some common integrations, but do not represent all available UID2 integration options. For a summary of all the integration options available, see [UID2 Integration Guides Overview](https://unifiedid.com/docs/guides/summary-guides).
```
The Sample Sites > The sample sites

genwhittTTD commented on i18n/ja/docusaurus-plugin-content-docs/current/guides/summary-guides.md:
```diff
@@ -31,6 +31,8 @@ import Link from '@docusaurus/Link';
 - [Prebid Integrations](#prebid-integrations)
 - [Google Ad Manager Integrations](#google-ad-manager-integrations)
 
+To explore live, working examples of UID2 implementations with source code, see [UID2 Integration Samples](../ref-info/integration-sample-sites.md).
```
Don't add fragments of English into the Japanese files. I didn't tell you this. But just leave it. The translator will pick it up, don't worry.

If we add bits in, they could be forgotten. It's just not the process. A whole page, we have to do it because of the broken out-links problem. Code and headings we can do because they're in English anyway. But not strings of text in translated files.

eiman-eltigani-ttd replied:
So I should leave this as is or remove the commas?

genwhittTTD replied:
@eiman-eltigani-ttd Remove line 34 which  you added. So, this file will not appear in the PR, it isn't changed. The translator will make the change.

genwhittTTD replied:
@eiman-eltigani-ttd this one is still open.

eiman-eltigani-ttd replied:
Fixed!

genwhittTTD reviewed (changes requested):
Thanks Eiman and looks a lot better. Few things to fix. Some things (like matching titles in link destinations, no code tags in links) I didn't mark all instances.

Will need one more pass but hopefully a quick one.

One substantive query is re including the hashing tool... it changes the nature of the page. Might need input from AS on that one. Your call. LMK if you need anything.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,101 @@
+
+### Web Integrations
+
+This section summarizes the sample integrations available for publishers who want to integrate UID2 directly into their websites. For a full summary of integration options for websites, see [Web Integration Overview](https://unifiedid.com/docs/guides/integration-options-publisher-web).
```
Destination copy is:

Publisher Web Integration Overview

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,101 @@
+
+This page lists all official UID2 sample implementations highlighting common integration use cases, along with links to live demo sites, source code, and related documentation. Use this page to quickly identify which sample matches your needs and explore working examples.
+
+### Web Integrations
```
I'm sorry I didn't notice this before. The heading levels are wrong. Title is h1 (one hash mark). What you have as ### should be ##. And what you have as #### should be ###. 
If you do that, the individual entries will show up on the right sidebar also.

I won't mark them all... this is a fix for throughout the doc.

The expanded right sidebar will increase usability, too.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,101 @@
+
+#### UID2 Server-Only (Server-Side)
+
+This sample is for publishers who want all UID2 operations to occur on the server-side, offering maximum control, security, and flexibility without relying on a client-side SDK.
```
on the server-side > on the server side. (Only hyphenated when used as a modifier)

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,101 @@
+
+- Site: [UID2 Publisher Client-Side Integration Example using UID2 JavaScript SDK](https://cstg-integ.uidapi.com/)
+- Code: [uid2-web-integrations/examples/cstg](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/cstg)
+- Docs: [Client-Side Integration Guide for JavaScript](https://unifiedid.com/docs/guides/integration-javascript-client-side)
```
Two things, on all docs links:

1) Doc, not docs, since we're linking to only one document.

2) I missed that we're including full URLs. On anything that's in the Docusaurus site, it should be a relative URl. Otherwise it opens up the destination in a new browser window... and my local build opens up the published site.

Same two edits for all docs links.

In this instance:

`Docs: [Client-Side Integration Guide for JavaScript](https://unifiedid.com/docs/guides/integration-javascript-client-side)`

> 

`Doc [Client-Side Integration Guide for JavaScript](../guides/integration-javascript-client-side)`

eiman-eltigani-ttd replied:
Is this required? We wanted it to open to a destination to keep it consistent with the rest and in general makes it easier for user to read so they dont have keep losing their place in the doc. I updated it for now but lmk if this is something we can revert.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,101 @@
+
+- Site: [UID2 Publisher Client-Side Integration Example using UID2 JavaScript SDK, Secure Signals](https://secure-signals-client-side-integ.uidapi.com/)
+- Code: [uid2-web-integrations/examples/google-secure-signals-integration/client_side](https://github.com/IABTechLab/uid2-web-integrations/tree/main/examples/google-secure-signals-integration/client_side)
+- Docs: [Secure Signals Integration Guide](https://unifiedid.com/docs/guides/integration-google-ss)
```
Same two comments re docs > doc and relative URL.

But also, link copy should match destination. In this case the title should be:

Google Ad Manager Secure Signals Integration Guide

Two (or more?)  instances of this link. Please be sure to fix any instances and, generally, link copy should match the destination exactly when it's the title of a document.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,101 @@
+- Docs: [Normalization and Encoding](https://unifiedid.com/docs/getting-started/gs-normalization-encoding)
+
+:::note
+The sample sites in this file highlight some common integrations, but do not represent all available UID2 integration options. For a summary of all the integration options available, see [UID2 Integration Guides Overview](https://unifiedid.com/docs/guides/summary-guides).
```
Destination title is incorrect. Should be: UID2 Integration Guides: Summary

genwhittTTD reviewed (changes requested):
Looking much better. Some inconsistencies still plus I noticed a few key things I missed before. LMK if any questions.

genwhittTTD commented on docs/ref-info/integration-sample-sites.md:
```diff
@@ -0,0 +1,105 @@
+
+This section includes a sample page for generating UID2 tokens to be passed by Prebid.js in the RTB bidstream. For details, see [UID2 Integration Overview for Prebid](../guides/integration-prebid).
+
+### Client-Side Integration with Prebid.js
```
OK.

genwhittTTD reviewed (commented):
One small extra edit. Couple of open discussions which are your call. One remove edit on integration-sample-sites.md, update the copy in the Japanese if needed, and that's it.

With those fixes we're good to go. I don't need to see it again.

eiman-eltigani-ttd commented:
Summary of most recent changes:
- Added both publisher and advertiser/dp links for web-integration overview. Lmk if it makes sense to remove from the DSP tab because it is now currently available in all tabs
- Updated doc references to external sites

genwhittTTD reviewed (approved):
LGTM thx.
